haystack/preview/components/file_converters/markdown.py

```diff
@@ -0,0 +1,119 @@
+import logging
+import re
+from pathlib import Path
+from typing import Dict, List, Optional, Tuple, Union, Any
+
+from tqdm import tqdm
+
+from haystack.preview import Document, component
+from haystack.preview.lazy_imports import LazyImport
+
+with LazyImport("Run 'pip install beautifulsoup4 markdown python-frontmatter'") as markdown_conversion_imports:
+    import frontmatter
+    from bs4 import BeautifulSoup, NavigableString
+    from markdown import markdown
+
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class MarkdownToTextDocument:
+    """
+    Converts a Markdown file into a text Document.
+    """
+
+    def __init__(
+        self,
+        remove_code_snippets: bool = True,
+        extract_headlines: bool = False,
+        add_frontmatter_to_meta: bool = False,
+        progress_bar: bool = True,
+    ):
+        """
+        :param remove_code_snippets: Whether to remove snippets from the markdown file. Defaults to True.
+        :param extract_headlines: Whether to extract headings from the markdown file. Defaults to False.
+        :param add_frontmatter_to_meta: Whether to add the contents of the frontmatter to `meta`. Defaults to False.
+        :param progress_bar: Show a progress bar for the conversion.
+        """
+        markdown_conversion_imports.check()
+
+        self.remove_code_snippets = remove_code_snippets
+        self.extract_headlines = extract_headlines
+        self.add_frontmatter_to_meta = add_frontmatter_to_meta
+        self.progress_bar = progress_bar
+
+    @component.output_types(documents=List[Document])
+    def run(self, paths: List[Union[str, Path]], metadata: Optional[List[Union[Dict, Any]]] = None):
```

Comment:
I would like this component to support also `ByteStream`s.

Please bear with me on this aspect. I'm figuring out how to support this properly...

Reply:
Update: you can do something similar to `PyPDFToDocument`:
https://github.com/deepset-ai/haystack/blob/3ff79d0307db167ce128ac235ac6e5215607dea6/haystack/preview/components/file_converters/pypdf.py#L30